client/components/at-internet/at-internet.config.js

```diff
@@ -14,49 +14,21 @@ angular.module("managerApp")
     });
 
 })
-.config(["$provide", function ($provide) {
+.run(($cookies, atInternet, TRACKING, TARGET, OvhApiMe) => {
     "use strict";
 
-
-    $provide.decorator("atInternet", function ($delegate, $q, $cookies, OvhApiMe, TRACKING, TARGET) {
-        var delegateTrackPage = $delegate.trackPage;
-        var isDefaultConfigurationSet = false;
-        var trackPageRequestArgumentStack = [];
-
-        //Decorate trackPage to stack requests until At-internet default configuration is set
-        $delegate.trackPage = function () {
-            if (isDefaultConfigurationSet) {
-                delegateTrackPage.apply($delegate, arguments);
-            } else {
-                trackPageRequestArgumentStack.push(arguments);
-            }
-        };
-
-        OvhApiMe.Lexi().get().$promise.then(function (me) {
-            var settings = angular.copy(TRACKING[TARGET].config);
-
-            var referrerSite = $cookies.get("OrderCloud");
-            if (referrerSite) {
-                settings.referrerSite = referrerSite;
-            }
-
-            settings.countryCode = me.country;
-            settings.currencyCode =  me.currency && me.currency.code;
-            $delegate.setDefaults(settings);
-
-            isDefaultConfigurationSet = true;
-
-            _.forEach(trackPageRequestArgumentStack, function (trackPageArguments) {
-                delegateTrackPage.apply($delegate, trackPageArguments);
-            });
-
-        }).catch(function (err) {
-            $delegate.trackPage = function () {};
-            trackPageRequestArgumentStack = [];
-            $delegate.setEnabled(false);
-            return $q.reject(err);
+    let config = TRACKING[TARGET].config;
+    const referrerSite = $cookies.get("OrderCloud");
+
+    if (referrerSite) {
+        config.referrerSite = referrerSite;
```

Comment:
why not config.referrerSite = referrerSite at all time?  Even if referrerSite is undefined or null?

Reply:
I didn't want to polluate data sent to ng-at-internet. 
Moreover ng-at-internet has been built, to be use with some "general" use case, so I cared about not handle this mechanics inside the lib.